### DISCUSSION TITLE
RuntimeError - Transform

### DISCUSSION THREAD

**Discussion by HripsimeS** [Project-MONAI/MONAI#3817] [Q&A]
Dear MONAI community,

I am trying to test the following notebook for 3D Unet segmentation
https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/spleen_segmentation_3d_lightning.ipynb

On the stage of "Run the training", the process is interrupting in the middle with following error message "RuntimeError: applying transform <monai.transforms.spatial.dictionary.Spacingd object at 0x0000019CAC2BB970>". I tried to reduce "num_workers" and "batch_size", but it did not help. Can you please provide some solution for this error/bug. 

Look forward to hearing from you.

**Comment by wyli** [comment]
it seems it works fine with the google colab (with a gpu runtime) https://colab.research.google.com/github/Project-MONAI/tutorials/blob/master/3d_segmentation/spleen_segmentation_3d_lightning.ipynb, so it's probably your local environment issue, I'm converting this into a discussion. please follow the issue reporting template with more details so that we might be able to help.

**Reply by HripsimeS** [reply]
Thank you @wyli. Now I am trying to optimize the same notebook I mentioned above for my dataset, 3D images CT images with their labels. But unfortunately I am getting the following error after execution of "Execute a typical PyTorch training process" part. It seems that the running is having an error on loss_function calculation on line 24 and eventually I received a "RuntimeError: index 3 is out of bounds for dimension 1 with size 2". Do you have some ideas what is going on and how to fix this error? Thanks a lot in advance!


RuntimeError                              Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_19120/3283586599.py in <module>
     22         optimizer.zero_grad()
     23         outputs = model(inputs)
---> 24      loss = loss_function(outputs, labels)
     25         loss.backward()
     26         optimizer.step()

~\anaconda3\envs\abys\lib\site-packages\torch\nn\modules\module.py in _call_impl(self, *input, **kwargs)
   1100         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
   1101                 or _global_forward_hooks or _global_forward_pre_hooks):
-> 1102             return forward_call(*input, **kwargs)
   1103         # Do not call functions when jit is used
   1104         full_backward_hooks, non_full_backward_hooks = [], []

~\anaconda3\envs\abys\lib\site-packages\monai\losses\dice.py in forward(self, input, target)
    144                 warnings.warn("single channel prediction, `to_onehot_y=True` ignored.")
    145             else:
--> 146                 target = one_hot(target, num_classes=n_pred_ch)
    147 
    148         if not self.include_background:

~\anaconda3\envs\abys\lib\site-packages\monai\networks\utils.py in one_hot(labels, num_classes, dtype, dim)
     89 
     90     o = torch.zeros(size=sh, dtype=dtype, device=labels.device)
---> 91     labels = o.scatter_(dim=dim, index=labels.long(), value=1)
     92 
     93     return labels

RuntimeError: index 3 is out of bounds for dimension 1 with size 2

**Reply by wyli** [reply]
looks like the network has `out_channels=2` but the input `labels` have more than 2 classes..
